architecture/file_structure.rst

```diff
@@ -1,186 +1,288 @@
+    * - libdl/include/
+      - Include files for libdl
 
-    * - /mm
-      - Include files exposing API for /core/arch/arm/mm files
+    * - libmbedtls/
+      - Mbedtls crypto library
 
-    * - /rom
-      - Old ROM files that should be removed before going public
+    * - libmbedtls/core/
+      - Glue code only compiled with core to connect with the core internal
+        <crypto/crypto.h> API.
 
-    * - /sm
-      - Include files exposing API for Secure Monitor
+    * - libmbedtls/include/
+      - Include files with configuration of mbedtls
 
-/core/include
-*************************
-.. list-table:: Structure of /core/include
-    :header-rows: 1
-    :widths: 1 5
+    * - libmbedtls/mbedtls/
+      - Top directory of the imported mbedtls source tree
 
-    * - Directory
-      - Description
+    * - libmbedtls/mbedtls/include/
+      - Mbedtls include files
 
-    * - /drivers
-      - Include files exposing API for /core/drivers files
+    * - libmbedtls/mbedtls/library/
+      - Mbedtls implementation
 
-    * - /dt-bindings
-      - Include files for the device tree bindings
+    * - libunw/
+      - Unwind library 
 
-/core/lib/lib{crypto,sla}
-*************************
-.. list-table:: Structure of /core/lib/lib{crypto,sla}
-    :header-rows: 1
-    :widths: 1 5
+    * - libunw/include/
+      - Include files for libunwnd
 
-    * - Directory
-      - Description
+    * - libutee/
+      - Libutee which provide the implementation of TEE Internal Core API.
+
+    * - libutee/arch/
+      - Architecure specific implementation
+
+    * - libutee/include/
+      - Include files relating to libutee and the header files for
+        TEE Internal Core API
+
+    * - libutils/
+      - The reduced "libc" of OP-TEE
+
+    * - libutils/ext/
+      - Extensions to a standard libc
+
+    * - libutils/ext/arch/
+      - Architecture specific implmementation of the extensions
+
+    * - libutils/ext/include/
+      - Include files relating to the extensions
+
+    * - libutils/isoc/
+      - A subset of ISOC
 
-    * - /
-      - Source files for the library
+    * - libutils/isoc/arch/
+      - Architecture specific
 
-    * - /include  
-      - Include files exposing the API of the library
+    * - libutils/isoc/include/
+      - Header files relating to the provided subset of ISOC
 
-/lib/libutils
+    * - libutils/isoc/newlib/
+      - Routines imported from newlib
+
+
+ta/
 *************
-.. list-table:: Structure of /lib/libutils
+.. list-table::
     :header-rows: 1
     :widths: 1 5
 
     * - Directory
       - Description
 
-    * - /
-      - Source file for the library
+    * - trusted_keys
+      - Trusted key TA
+
+    * - trusted_keys/include
+      - Header file of the ABI provided by the trusted key TA
+
+    * - arch
+      - Architecture specific files needed to compile a TA
+
+    * - mk
+      - Makefile includes needed to build TAs and the TA dev kit
+
+    * - avb
+      - TA to support AVB (Android Verified Boot)
```

Comment:
Worth pointing out that it's AVB 2.0 that we adhere to?

Reply:
This might age better without such details.

Reply:
Sure